### PR TITLE
Release Notes for Jan. 2024 Prerelease SDK

### PR DESCRIPTION
**Pull request by mikehoffms**
Rendered article section for review:
https://review.learn.microsoft.com/microsoft-edge/webview2/release-notes?branch=pr-en-us-3023&tabs=dotnetcsharp#102357-prerelease

AB#48553403

### CODE REVIEW

File: microsoft-edge/webview2/release-notes.md [hunk 1]
```diff
@@ -144,6 +144,372 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2profile?view=webview2-winrt-1.0.2357-prerelease&preserve-view=true)
+    * [CoreWebView2Profile.GetBrowserExtensionsAsync Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2profile?view=webview2-winrt-1.0.2357-prerelease&preserve-view=true#getbrowserextensionsasync)
+
+##### [Win32/C++](#tab/win32cpp)
+N/A
+
+---
+
+<!-- ---------- -->
+*  Added support for `WebResourceRequested` for workers which allows setting filters in order to receive `WebResourceRequested` events for service workers, shared workers, and different origin iframes.
```
Comment (peiche-jessica) [inline microsoft-edge/webview2/release-notes.md:464]: The enum CoreWebView2WebResourceRequestSourceKinds / COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS is in stable too if we'd like to call them out.

File: microsoft-edge/webview2/release-notes.md [hunk 2]
```diff
@@ -144,6 +144,381 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+
+---
+
+<!-- ------------------------------ -->
+#### Promotions
+
+The following APIs have been promoted from Experimental to Stable in this Prerelease SDK.
+
+
+<!-- todo: add enum
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:266]: @victorhuangwq 
todo: add enum
CoreWebView2WebResourceRequestSourceKinds - 5 hits below; copypaste
COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS - 0 hits below (why 0? typo?)

you or i can do

File: microsoft-edge/webview2/release-notes.md [hunk 3]
```diff
@@ -144,6 +144,381 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
 <!-- end of Jan 2024 Prerelease SDK -->
 
 
+<!-- ====================================================================== -->
+## 1.0.2357-prerelease
+
+<!-- todo: update version number & date -->
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:150]: todo: update version number & date (Prerelease) @victorhuangwq

File: microsoft-edge/webview2/release-notes.md (hunk 2 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: i removed this todo comment and in the correct spot, started adding 3 enum links, but can't find URL, so added 3 todo notes there

File: microsoft-edge/webview2/release-notes.md (hunk 1 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:469]: started adding 3 enum links, url not yet found so entered 3 todos/3 PR Comments

File: microsoft-edge/webview2/release-notes.md [hunk 4]
```diff
@@ -144,6 +144,386 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* `CoreWebView2` Class:
+   * [CoreWebView2.AddWebResourceRequestedFilter(RequestSourceKinds) Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.addwebresourcerequestedfilter?view=webview2-dotnet-1.0.2357-prerelease&preserve-view=true)
+   * [CoreWebView2.RemoveWebResourceRequestedFilter(RequestSourceKinds) Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.removewebresourcerequestedfilter?view=webview2-dotnet-1.0.2357-prerelease&preserve-view=true)
+
+* [CoreWebView2WebResourceRequestedEventArgs Class](/dotnet/api/microsoft.web.webview2.core.corewebview2webresourcerequestedeventargs?view=webview2-dotnet-1.0.2357-prerelease&preserve-view=true)
+
+<!-- todo: add enum url: -->
+* [CoreWebView2WebResourceRequestSourceKinds Enum]
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:480]: add enum URL or approx URL (.NET).  not found yet in Ref docs
@peiche-jessica @victorhuangwq

File: microsoft-edge/webview2/release-notes.md [hunk 5]
```diff
@@ -144,6 +144,386 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2.AddWebResourceRequestedFilter(requestSourceKinds) Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?view=webview2-winrt-1.0.2357-prerelease&preserve-view=true#addwebresourcerequestedfilter)
+* [CoreWebView2.RemoveWebResourceRequestedFilter(requestSourceKinds) Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2?view=webview2-winrt-1.0.2357-prerelease&preserve-view=true#removewebresourcerequestedfilter)
+* [CoreWebView2WebResourceRequestedEventArgs Class](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2webresourcerequestedeventargs?view=webview2-winrt-1.0.2357-prerelease&preserve-view=true)
+
+<!-- todo: add enum url: -->
+* [CoreWebView2WebResourceRequestSourceKinds Enum]
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:489]: add enum URL or approx URL (WinRT).  not found yet in Ref docs
@peiche-jessica @victorhuangwq

File: microsoft-edge/webview2/release-notes.md [hunk 6]
```diff
@@ -144,6 +144,386 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+    * [ICoreWebView2_22::RemoveWebResourceRequestedFilterWithRequestSourceKinds](/microsoft-edge/webview2/reference/win32/icorewebview2_22?view=webview2-1.0.2357-prerelease&preserve-view=true#removewebresourcerequestedfilterwithrequestsourcekinds)
+
+* [ICoreWebView2WebResourceRequestedEventArgs2](/microsoft-edge/webview2/reference/win32/icorewebview2webresourcerequestedeventargs2?view=webview2-1.0.2357-prerelease&preserve-view=true)
+    * [ICoreWebView2WebResourceRequestedEventArgs2::get_RequestedSourceKind](/microsoft-edge/webview2/reference/win32/icorewebview2webresourcerequestedeventargs2?view=webview2-1.0.2357-prerelease&preserve-view=true#get_requestedsourcekind)
+    * [ICoreWebView2WebResourceRequestedEventArgs2::Invoke](/microsoft-edge/webview2/reference/win32/icorewebview2webresourcerequestedeventargs2?view=webview2-1.0.2357-prerelease&preserve-view=true#invoke)
+
+<!-- todo: add enum url: -->
+* [COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS enum]
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:502]: add enum URL or approx URL (Win32).  not found yet in Ref docs
@peiche-jessica @victorhuangwq

File: microsoft-edge/webview2/release-notes.md (hunk 5 above)
Comment (peiche-jessica) [reply microsoft-edge/webview2/release-notes.md:489]: It should be something like `/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2webresourcerequestsourcekinds?view=webview2-winrt-1.0.2194-prerelease`

File: microsoft-edge/webview2/release-notes.md (hunk 6 above)
Comment (peiche-jessica) [reply microsoft-edge/webview2/release-notes.md:502]: It should be something like `/microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-win32-preview&branch=pr-en-us-78#corewebview2_web_resource_request_source_kinds`

File: microsoft-edge/webview2/release-notes.md (hunk 4 above)
Comment (peiche-jessica) [reply microsoft-edge/webview2/release-notes.md:480]: It should be something like `/dotnet/api/microsoft.web.webview2.core.corewebview2webresourcerequestsourcekinds?view=webview2-dotnet-1.0.2194-prerelease`

File: microsoft-edge/webview2/release-notes.md [hunk 7]
```diff
@@ -144,6 +144,386 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+<!-- ====================================================================== -->
+## 1.0.2357-prerelease
+
+<!-- todo: update version number & date -->
+
+Release Date: January 26, 2024
+
+[NuGet package for WebView2 SDK 1.0.2357-prerelease](https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.2357-prerelease)
+
+For full API compatibility, this Prerelease version of the WebView2 SDK requires the WebView2 Runtime that ships with Microsoft Edge version 122.0.2363.0 or higher.
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:156]:
```suggestion
For full API compatibility, this Prerelease version of the WebView2 SDK requires the WebView2 Runtime that ships with Microsoft Edge version 122.0.2357.0 or higher.
```
@peiche-jessica @victorhuangwq  - tech review this suggestion.  In previous Prerelease sections, this # is always same as prerelease's #.

File: microsoft-edge/webview2/release-notes.md (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:150]: keeping version as-is.  updated date

File: microsoft-edge/webview2/release-notes.md (hunk 6 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:502]:
```
* [COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_KINDS enum]
(/microsoft-edge/webview2/reference/win32/webview2-idl
?view=webview2-1.0.2357-prerelease&preserve-view=true
#corewebview2_web_resource_request_source_kinds)
```

File: microsoft-edge/webview2/release-notes.md (hunk 5 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:489]:
```
* [CoreWebView2WebResourceRequestSourceKinds Enum]
(/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2webresourcerequestsourcekinds
?view=webview2-winrt-1.0.2357-prerelease&preserve-view=true)
```

File: microsoft-edge/webview2/release-notes.md (hunk 4 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:480]:
```
* [CoreWebView2WebResourceRequestSourceKinds Enum]
(/dotnet/api/microsoft.web.webview2.core.corewebview2webresourcerequestsourcekinds
?view=webview2-dotnet-1.0.2357-prerelease&preserve-view=true)
```

File: microsoft-edge/webview2/release-notes.md [hunk 8]
```diff
@@ -144,6 +144,386 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
+    * [ICoreWebView2_22::AddWebResourceRequestedFilterWithRequestSourceKinds](/microsoft-edge/webview2/reference/win32/icorewebview2_22?view=webview2-1.0.2357-prerelease&preserve-view=true#addwebresourcerequestedfilterwithrequestsourcekinds)
+    * [ICoreWebView2_22::RemoveWebResourceRequestedFilterWithRequestSourceKinds](/microsoft-edge/webview2/reference/win32/icorewebview2_22?view=webview2-1.0.2357-prerelease&preserve-view=true#removewebresourcerequestedfilterwithrequestsourcekinds)
+
+* [ICoreWebView2WebResourceRequestedEventArgs2](/microsoft-edge/webview2/reference/win32/icorewebview2webresourcerequestedeventargs2?view=webview2-1.0.2357-prerelease&preserve-view=true)
+    * [ICoreWebView2WebResourceRequestedEventArgs2::get_RequestedSourceKind](/microsoft-edge/webview2/reference/win32/icorewebview2webresourcerequestedeventargs2?view=webview2-1.0.2357-prerelease&preserve-view=true#get_requestedsourcekind)
+    * [ICoreWebView2WebResourceRequestedEventArgs2::Invoke](/microsoft-edge/webview2/reference/win32/icorewebview2webresourcerequestedeventargs2?view=webview2-1.0.2357-prerelease&preserve-view=true#invoke)
```
Comment (mikehoffms) [inline microsoft-edge/webview2/release-notes.md:499]:
```suggestion
```
no Invoke is at 
https://review.learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2webresourcerequestedeventargs2?view=webview2-1.0.2357-prerelease&branch=pr-en-us-78&preserve-view=true#invoke

Comment (peiche-jessica) [reply microsoft-edge/webview2/release-notes.md:499]: Please delete. Thanks

Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: deleted

File: microsoft-edge/webview2/release-notes.md (hunk 7 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: latest commit adds hidden comment:
```
For full API compatibility, this Prerelease version of the WebView2 SDK requires 
the WebView2 Runtime that ships with Microsoft Edge version 122.0.2363.0 or 
higher.<!-- todo: differs from prerelease # this time? -->
```

File: microsoft-edge/webview2/release-notes.md [hunk 9]
```diff
@@ -144,6 +144,372 @@ The following APIs have been promoted from Experimental to Stable in this Prerel
...
 
 
+<!-- ====================================================================== -->
+## 1.0.2357-prerelease
+
+Release Date: January 30, 2024
+
+[NuGet package for WebView2 SDK 1.0.2357-prerelease](https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.2357-prerelease)
+
+For full API compatibility, this Prerelease version of the WebView2 SDK requires the WebView2 Runtime that ships with Microsoft Edge version 122.0.2363.0 or higher.<!-- todo: differs from prerelease # this time? -->
```
Comment (peiche-jessica) [inline microsoft-edge/webview2/release-notes.md:154]:
```suggestion
For full API compatibility, this Prerelease version of the WebView2 SDK requires the WebView2 Runtime that ships with Microsoft Edge version 122.0.2357.0 or higher.<!-- todo: differs from prerelease # this time? -->

```

@victorhuangwq Should this be 2357 instead?

**Review by peiche-jessica — APPROVED** [review]

File: microsoft-edge/webview2/release-notes.md (hunk 7 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]:
```
For full API compatibility, this Prerelease version of the WebView2 SDK requires 
the WebView2 Runtime that ships with Microsoft Edge version 122.0.2357.0 or higher.
```

File: microsoft-edge/webview2/release-notes.md (hunk 9 above)
Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md:154]:
```
For full API compatibility, this Prerelease version of the WebView2 SDK requires 
the WebView2 Runtime that ships with Microsoft Edge version 122.0.2357.0 or higher.
```

Comment (victorhuangwq) [reply microsoft-edge/webview2/release-notes.md:154]: yes this should be 2357.

Comment (mikehoffms) [reply microsoft-edge/webview2/release-notes.md]: pushed commit, building...

**Review by victorhuangwq — APPROVED** [review]
LGTM

**Pull request merged**